theano/tensor/subtensor.py

```diff
@@ -416,13 +422,27 @@ def conv(val):
                              conv(val.stop),
                              conv(val.step))
             else:
-                try:
-                    return get_scalar_constant_value(val)
-                except theano.tensor.NotScalarConstantError:
-                    if allow_partial:
-                        return val
+                if only_process_constants:
+                    if isinstance(val, Constant):
+                        if getattr(val.tag, 'unique_value', None) is not None:
```

Comment:
I would call get_scalar_constant_value() here instead of copy/pasting the code. In case this code change.